File: app/static/js/overview.js
function formatBytes(bytes) {
  if (!bytes) return "0 GB";
  const gb = bytes / (1024 ** 3);
  if (gb >= 1) return gb.toFixed(2) + " GB";
  const mb = bytes / (1024 ** 2);
  return mb.toFixed(1) + " MB";
}

async function loadOverview() {
  const res = await fetch("/api/overview");
  const data = await res.json();
  document.getElementById("stat-total-videos").textContent = data.total_videos;
  document.getElementById("stat-storage").textContent = formatBytes(data.total_storage_bytes);
  document.getElementById("stat-views").textContent = data.total_views;
  document.getElementById("stat-processing").textContent = data.processing_count;
  document.getElementById("stat-ready").textContent = data.ready_count;
  document.getElementById("stat-error").textContent = data.error_count;
}

loadOverview();
setInterval(loadOverview, 10000);

File: app/routes/overview.py
from fastapi import APIRouter
from app.database import videos_collection

router = APIRouter(prefix="/api/overview", tags=["overview"])


@router.get("")
async def get_overview():
    total_videos = await videos_collection.count_documents({})
    processing_count = await videos_collection.count_documents({"status": "PROCESSING"})
    ready_count = await videos_collection.count_documents({"status": "READY"})
    error_count = await videos_collection.count_documents({"status": "ERROR"})

    total_views = 0
    total_storage_bytes = 0
    async for doc in videos_collection.find({}, {"views": 1, "all_files": 1}):
        total_views += doc.get("views", 0)
        for f in doc.get("all_files", []):
            total_storage_bytes += f.get("size", 0)

    return {
        "total_videos": total_videos,
        "total_storage_bytes": total_storage_bytes,
        "total_views": total_views,
        "processing_count": processing_count,
        "ready_count": ready_count,
        "error_count": error_count,
    }
